scalding-core/src/main/scala/com/twitter/scalding/ExecutionContext.scala

```diff
@@ -52,6 +76,25 @@ trait ExecutionContext {
       }
       val flow = mode.newFlowConnector(withId).connect(flowDef)
 
+      if (flow.isInstanceOf[HadoopFlow]) {
+        val flowSteps = flow.asInstanceOf[HadoopFlow].getFlowSteps.toList
+        flowSteps.foreach(step => {
+          val baseFlowStep: BaseFlowStep[JobConf] = step.asInstanceOf[BaseFlowStep[JobConf]]
+
+          val descriptions: Seq[String] = {
+            for {
+              vertex <- baseFlowStep.getGraph.vertexSet.toList
+              if (vertex.isInstanceOf[Pipe])
```

Comment:
```scala
def getDesc(v: Any): Seq[String] = v match {
  case vPipe: Pipe => RichPipe.getPipeDescriptions(vPipe)
  case _ => List() // no descriptions
}
```
then call that rather than the test + cast (do it in one go and no explicit cast).